File: Application/Models/Order/OrderItemModel.cs
using Application.Models.Base;
using Application.Models.Product;
using Core.Enums;

namespace Application.Models.Order
{
    public class OrderItemModel: BaseModel
    {
        public string ProductId { get; set; }

        public ProductModel Product { get; set; }

        public string OrderId { get; set; }

        public OrderModel Order { get; set; }

        public Size ProductSize { get; set; }

        public uint Count { get; set; }
    }
}

File: Web/Client/src/app/modules/order/resources/models/order-item.model.ts
import { Size } from "src/app/modules/product/resources/models/product-size.enum";
import { Product } from "src/app/modules/product/resources/models/product.model";
import { Order } from "./order.model";

export interface OrderItem {
  id: string
  productId: string
  product: Product
  orderId?: string
  order?: Order
  productSize: Size
  count: number
  createdDate?: string
}
